extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -323,16 +334,26 @@ export class ResourceTypePickerDialog extends DialogBase {
 
 	private async installTools(): Promise<void> {
 		this._installToolButton.enabled = false;
-		let i: number = 0;
+		let tool: ITool;
 		try {
-			for (; i < this._tools.length; i++) {
-				if (this._tools[i].needsInstallation) {
+			const toolRequirements = this.getCurrentProvider().requiredTools;
+			for (let i: number = 0; i < toolRequirements.length; i++) {
```

Comment:
You changed the logic here now so that the index is from the toolRequirements array but didn't update all the dialog messages below (such as line 347). Make sure you go through and fix those to use the appropriate tool reference.

Reply:
Good catch. Will do the necessary update